cmd/agent/main.go

```diff
@@ -86,6 +86,8 @@ func main() {
 	}
 
 	if cfg.Integrations.Enabled {
+		// If a client auth is defined in TLS assume HTTPS/TLS
+		cfg.Integrations.SeverUsingTLS = cfg.Server.HTTPTLSConfig.ClientAuth != ""
```

Comment:
The server using TLS is dictated by it having a TLS key pair set, not if it's using client authentication: 

```suggestion
		cfg.Integrations.SeverUsingTLS = cfg.Server.HTTPTLSConfig.TLSCertPath != "" || cfg.Server.HTTPTLSConfig.TLSKeyPath != ""
```

See the logic [here](https://github.com/weaveworks/common/blob/da2819181d057aba9cca4ba79226d808ebf7a00f/server/server.go#L181) for where that check happens.

Reply:
Actually used the && instead of || as per the logic.